### PR TITLE
Add exports to package.json

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #1464

### CODE REVIEW

**Review by cjtenny — APPROVED** [review]
:goal_net:

**Comment by ljharb** [comment]
Note that adding this is a breaking change; the format you've used won't work in every node version, and I'm not sure why this improvement is needed given that the polyfill should theoretically be deprecated _very_ soon?

File: polyfill/package.json [hunk 1]
```diff
@@ -5,6 +5,10 @@
   "type": "commonjs",
   "main": "dist/index.js",
   "browser": "dist/index.umd.js",
+  "exports": {
+    "import": "./lib/index.mjs",
+    "require": "./dist/index.js"
+  },
```
Comment (ljharb) [inline polyfill/package.json:11]: this is what would be needed if you wanted to actually support every node version:
```suggestion
  "exports": {
     ".": [
       {
        "import": "./lib/index.mjs",
        "default": "./dist/index.js"
       },
       "./dist/index.js"
     ]
  },
```

but note that this is still a breaking change unless these are the only two files included in the package (which i can see they aren't from the "browser" field above)

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by ptomato** [comment]
> Note that adding this is a breaking change; the format you've used won't work in every node version, and I'm not sure why this improvement is needed given that the polyfill should theoretically be deprecated _very_ soon? (arguably, should have been deprecated a month ago, per prior agreements)

I don't agree that that precludes changes to leave it in a good place, but I didn't realize this was a breaking change.

**Comment by ljharb** [comment]
@ptomato "leaving it in a good place" is fine within a reasonable timeframe, ofc - adding new features tho, breaking or otherwise, seems a bit out of scope of that.

**Comment by littledan** [comment]
What's wrong with a breaking change on something at version 0.8?

**Comment by ptomato** [comment]
Certainly breaking changes are fine at version 0.8, but once I realized this was breaking, it seemed like just churn without a clear benefit, unlike the other breaking changes we've had.

**Pull request closed without merging**